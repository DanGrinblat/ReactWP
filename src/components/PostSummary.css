.post-summary {
	width: 300px;
}

.post-summary-text {
	color: black;
}
