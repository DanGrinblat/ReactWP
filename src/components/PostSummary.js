import "./PostSummary.css";
import React from "react";

class PostSummary extends React.Component {
	//Remove all HTML tags
	strip(html) {
		var tmp = document.createElement("div");
		tmp.innerHTML = html;
		return tmp.textContent || tmp.innerText || "";
	}

	truncate = (string, maxLength) => {
		if (string.length < maxLength) {
			return string;
		}

		var truncatedString = string.substr(0, maxLength);

		//Re-trim if we are in the middle of a word
		truncatedString = truncatedString.substr(
			0,
			Math.min(truncatedString.length, truncatedString.lastIndexOf(" "))
		);
		return truncatedString + " ...";
	};

	render() {
		const { title, excerpt, imageLink } = this.props.post;
		var excerptFormatted = this.strip(excerpt.rendered);
		excerptFormatted = this.truncate(excerptFormatted, 100);

		return (
			<div>
				<img src={imageLink} alt="" class="post-summary-image" />
				<div className="post-summary-text">
					<b>{title.rendered}</b>
					<div>{excerptFormatted}</div>
				</div>
			</div>
		);
	}
}

export default PostSummary;
