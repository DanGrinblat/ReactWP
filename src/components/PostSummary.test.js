import React from "react";
import { shallow } from "enzyme";
import PostSummary from "./PostSummary";

describe("PostSummary", () => {
	const post = {
		featured_media: null,
		imageLink: null,
		title: { rendered: "title" },
		excerpt: { rendered: "excerpt" }
	};

	const postSummary = shallow(<PostSummary post={post} />);

	it("should render correctly", () => {
		expect(postSummary).toMatchSnapshot();
	});

	describe("truncation method", () => {
		it("should trim a string to the last whole word and add ellipses", () => {
			expect(postSummary.instance().truncate("test string", 6)).toEqual(
				"test ..."
			);
		});

		it("should return the string if its length is less than the specified length", () => {
			expect(postSummary.instance().truncate("test string", 20)).toEqual(
				"test string"
			);
		});
	});

	describe("strip method", () => {
		it("should remove all html tags from a string", () => {
			const sampleTest = "<div>test</div>";
			expect(postSummary.instance().strip(sampleTest)).toEqual("test");
		});

		it("should return an empty string if it receives an empty string", () => {
			expect(postSummary.instance().strip("")).toEqual("");
		});
	});
});
