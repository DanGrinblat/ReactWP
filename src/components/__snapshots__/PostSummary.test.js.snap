// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PostSummary should render correctly 1`] = `
<div>
  <img
    alt=""
    class="post-summary-image"
    src={null}
  />
  <div
    className="post-summary-text"
  >
    <b>
      title
    </b>
    <div>
      excerpt
    </div>
  </div>
</div>
`;
